vnext/Microsoft.ReactNative/TurboModulesProvider.cpp

```diff
@@ -166,7 +166,16 @@ class TurboModuleImpl : public facebook::react::TurboModule {
                             argWriter,
                             [promise, &runtime](const IJSValueWriter &writer) {
                               auto result = writer.as<JsiWriter>()->MoveResult();
-                              promise->resolve(result);
+                              if (result.isObject()) {
+                                auto resultArrayObject = result.getObject(runtime);
+                                VerifyElseCrash(resultArrayObject.isArray(runtime));
+                                auto resultArray = resultArrayObject.getArray(runtime);
+                                VerifyElseCrash(resultArray.length(runtime) == 1);
```

Comment:
is it not legal to return an array of multiple objects?

Reply:
This is guaranteed by the serialization code, when using `ReactPromise<Something>::Resolve`, there is always an array with the object that is passed to `Resolve`.